### PR TITLE
r/datasync_location_efs - add access point args

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Output from acceptance testing:

```
$ make testacc TESTS=TestAccDataSyncLocationEFS_ PKG=datasync
--- PASS: TestAccDataSyncLocationEFS_disappears (230.13s)
--- PASS: TestAccDataSyncLocationEFS_access (256.13s)
--- PASS: TestAccDataSyncLocationEFS_subdirectory (256.25s)
--- PASS: TestAccDataSyncLocationEFS_basic (256.69s)
--- PASS: TestAccDataSyncLocationEFS_tags (625.30s)
```

### CODE REVIEW

File: .changelog/25182.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+```release-note:enhancement
+resource/aws_datasync_location_efs: Add `access_point_arn`, `file_system_access_role_arn`, and `in_transit_cncryption` arguments
```
Comment (mmichaelb) [inline .changelog/25182.txt:2]: Shouldnt it be called `in_transit_encryption`?

File: internal/service/datasync/location_efs.go [hunk 2]
```diff
@@ -57,7 +66,19 @@ func ResourceLocationEFS() *schema.Resource {
...
-				ValidateFunc: validation.NoZeroValues,
+				ValidateFunc: verify.ValidARN,
+			},
+			"file_system_access_role_arn": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: verify.ValidARN,
+			},
+			"in_transit_cncryption": {
```
Comment (mmichaelb) [inline internal/service/datasync/location_efs.go:77]: same here, see https://github.com/hashicorp/terraform-provider-aws/pull/25182/files#r893261213

File: internal/service/datasync/location_efs.go [hunk 3]
```diff
@@ -99,10 +120,22 @@ func resourceLocationEFSCreate(d *schema.ResourceData, meta interface{}) error {
...
 
+	if v, ok := d.GetOk("access_point_arn"); ok {
+		input.AccessPointArn = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("file_system_access_role_arn"); ok {
+		input.FileSystemAccessRoleArn = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("in_transit_cncryption"); ok {
```
Comment (mmichaelb) [inline internal/service/datasync/location_efs.go:131]: see same here, see https://github.com/hashicorp/terraform-provider-aws/pull/25182/files#r893261213

File: internal/service/datasync/location_efs.go [hunk 4]
```diff
@@ -141,11 +174,14 @@ func resourceLocationEFSRead(d *schema.ResourceData, meta interface{}) error {
...
 	if err := d.Set("ec2_config", flattenEC2Config(output.Ec2Config)); err != nil {
-		return fmt.Errorf("error setting ec2_config: %s", err)
+		return fmt.Errorf("error setting ec2_config: %w", err)
 	}
 
 	d.Set("subdirectory", subdirectory)
 	d.Set("uri", output.LocationUri)
+	d.Set("access_point_arn", output.AccessPointArn)
+	d.Set("file_system_access_role_arn", output.FileSystemAccessRoleArn)
+	d.Set("in_transit_cncryption", output.InTransitEncryption)
```
Comment (mmichaelb) [inline internal/service/datasync/location_efs.go:184]: see https://github.com/hashicorp/terraform-provider-aws/pull/25182/files#r893261213

File: internal/service/datasync/location_efs_test.go [hunk 5]
```diff
@@ -51,6 +52,34 @@ func TestAccDataSyncLocationEFS_basic(t *testing.T) {
...
+		ErrorCheck:        acctest.ErrorCheck(t, datasync.EndpointsID),
+		ProviderFactories: acctest.ProviderFactories,
+		CheckDestroy:      testAccCheckLocationEFSDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccLocationEFSConfigAccess(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckLocationEFSExists(resourceName, &locationEfs1),
+					resource.TestCheckResourceAttrPair(resourceName, "access_point_arn", "aws_efs_access_point.test", "arn"),
+					resource.TestCheckResourceAttr(resourceName, "in_transit_cncryption", "TLS1_2"),
```
Comment (mmichaelb) [inline internal/service/datasync/location_efs_test.go:70]: see https://github.com/hashicorp/terraform-provider-aws/pull/25182/files#r893261213

File: internal/service/datasync/location_efs_test.go [hunk 6]
```diff
@@ -322,3 +338,22 @@ resource "aws_datasync_location_efs" "test" {
...
+func testAccLocationEFSConfigAccess() string {
+	return testAccLocationEFSBaseConfig() + `
+resource "aws_efs_access_point" "test" {
+  file_system_id = aws_efs_file_system.test.id
+}
+
+resource "aws_datasync_location_efs" "test" {
+  efs_file_system_arn   = aws_efs_mount_target.test.file_system_arn
+  access_point_arn      = aws_efs_access_point.test.arn
+  in_transit_cncryption = "TLS1_2"
```
Comment (mmichaelb) [inline internal/service/datasync/location_efs_test.go:351]: see https://github.com/hashicorp/terraform-provider-aws/pull/25182/files#r893261213

File: website/docs/r/datasync_location_efs.html.markdown [hunk 7]
```diff
@@ -31,8 +31,11 @@ resource "aws_datasync_location_efs" "example" {
 
 The following arguments are supported:
 
+* `access_point_arn` - (Optional) Specifies the Amazon Resource Name (ARN) of the access point that DataSync uses to access the Amazon EFS file system.
 * `ec2_config` - (Required) Configuration block containing EC2 configurations for connecting to the EFS File System.
 * `efs_file_system_arn` - (Required) Amazon Resource Name (ARN) of EFS File System.
+* `file_system_access_role_arn` - (Optional)  Specifies an Identity and Access Management (IAM) role that DataSync assumes when mounting the Amazon EFS file system.
+* `in_transit_cncryption` - (Optional) Specifies whether you want DataSync to use TLS encryption when transferring data to or from your Amazon EFS file system. Valid values are `NONE` and `TLS1_2`.
```
Comment (mmichaelb) [inline website/docs/r/datasync_location_efs.html.markdown:38]: see https://github.com/hashicorp/terraform-provider-aws/pull/25182/files#r893261213

File: internal/service/datasync/location_efs.go [hunk 8]
```diff
@@ -42,13 +48,16 @@ func ResourceLocationEFS() *schema.Resource {
...
+								Type:         schema.TypeString,
+								ValidateFunc: verify.ValidARN,
+							},
 						},
 						"subnet_arn": {
 							Type:         schema.TypeString,
 							Required:     true,
 							ForceNew:     true,
-							ValidateFunc: validation.NoZeroValues,
+							ValidateFunc: verify.ValidARN,
```
Comment (mmichaelb) [inline internal/service/datasync/location_efs.go:60]: these are changes to code logic not mentioned in the pr (as well as e.g. the %s/%w replacements) - I am not quite sure if they should be included within this pr

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccDataSyncLocationEFS_' PKG=datasync ACCTEST_PARALLELISM=3
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/datasync/... -v -count 1 -parallel 3  -run=TestAccDataSyncLocationEFS_ -timeout 180m
=== RUN   TestAccDataSyncLocationEFS_basic
=== PAUSE TestAccDataSyncLocationEFS_basic
=== RUN   TestAccDataSyncLocationEFS_accessPointARN
=== PAUSE TestAccDataSyncLocationEFS_accessPointARN
=== RUN   TestAccDataSyncLocationEFS_disappears
=== PAUSE TestAccDataSyncLocationEFS_disappears
=== RUN   TestAccDataSyncLocationEFS_subdirectory
=== PAUSE TestAccDataSyncLocationEFS_subdirectory
=== RUN   TestAccDataSyncLocationEFS_tags
=== PAUSE TestAccDataSyncLocationEFS_tags
=== CONT  TestAccDataSyncLocationEFS_basic
=== CONT  TestAccDataSyncLocationEFS_subdirectory
=== CONT  TestAccDataSyncLocationEFS_disappears
--- PASS: TestAccDataSyncLocationEFS_subdirectory (168.79s)
=== CONT  TestAccDataSyncLocationEFS_accessPointARN
--- PASS: TestAccDataSyncLocationEFS_accessPointARN (158.08s)
=== CONT  TestAccDataSyncLocationEFS_tags
--- PASS: TestAccDataSyncLocationEFS_disappears (340.11s)
--- PASS: TestAccDataSyncLocationEFS_basic (348.77s)
--- PASS: TestAccDataSyncLocationEFS_tags (173.02s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/datasync	514.560s
```

**Comment by ewbankkit** [comment]
@DrFaust92 Thanks for the contribution :tada: :clap:.

**Pull request merged**